orderly_web/config.py

```diff
@@ -97,13 +113,24 @@ def config_value(data, path, data_type, is_optional):
 
     expected = {"string": str,
                 "integer": int,
-                "boolean": bool}
+                "boolean": bool,
+                "dict": dict}
     if type(data) is not expected[data_type]:
         raise ValueError("Expected {} for {}".format(
             data_type, ":".join(path)))
     return data
 
 
+# TODO: once we have support for easily overriding parts of
+# configuration, this can be made better with respect to optional
+# values (e.g., if url is present other keys are required).
+def config_vault(data, path):
+    url = config_string(data, path + ["addr"], True)
```

Comment:
Is this really optional though? Looks like `start` calls `cfg.resolve_secrets` which will raise an Exception if there's no vault url provided.

Reply:
It is optional, and that is what happens when the (admittedly now-misnamed `complete`) configuration is used.

`cfg.resolve_secrets` calls `vault.resolve_secrets` and that only tries to use the vault if a string matching `VAULT:<path>:<key>` is found.

Reply:
Ah ok, I read `vault.client()` as throwing an error, but I see now it's returning a sort of dummy object that throws an error if you attempt to use it.

Reply:
yeah, exactly - any field or method used on that object will throw an error